### PR TITLE
IODA2 converter for concatenated raob BUFR data from LDM server

### LINKED ISSUE #722: create IODA2 converter for concatenated raob BUFR data from LDM server
## Description
create an ecCodes based IODA2 converter for raob BUFR files from an LDM server

## Requirements
this requires raob BUFR data and the ecCodes library and the python ecCodes interface

## Acceptance Criteria (Definition of Done)
successful production of IODA2 input file 

## Dependencies
ecCodes must be an option as well as the associated python interface

### PR DESCRIPTION
**Pull request by srherbener**
## Description

This PR adds in a bufr converter that handles concatenated raob data from the LDM server.

### Issue(s) addressed

- fixes #722

## Acceptance Criteria (Definition of Done)

The converter can create a ioda (raob sonde) test file that ufo can process successfully.

## Dependencies

Need eccodes access in containers and HPC platforms

## Impact

None

### CODE REVIEW

**Comment by fabiolrdiniz** [comment]
The list of tests failing here are as follow:
```
	  7 - iodaconv_metar_coding_norms (Failed)
	 72 - test_iodaconv_generic_bufr_raob (Failed)
	 73 - test_iodaconv_generic_bufr_raob_multi_files (Failed)
```

- `iodaconv_metar_coding_norms` is complaining about the norms in `metar_csv2ioda.py` and `decode_bufr_LDM_raob.py`
- `test_iodaconv_generic_bufr_raob` and `test_iodaconv_generic_bufr_raob` are complaining that `eccodes` is not available.

**Comment by fabiolrdiniz** [comment]
> * `iodaconv_metar_coding_norms` is complaining about the norms in `metar_csv2ioda.py` and decode_bufr_LDM_raob

The norms failing under the `iodaconv_metar_coding_norms` test are related to the `decode_bufr_LDM_raob` program only. I would suggest renaming the `iodaconv_metar_coding_norms` to `iodaconv_conventional_coding_norms` since this test will be testing all `*.py` inside the `conventional` folder.

File: src/conventional/decode_bufr_LDM_raob.py [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+import sys
+import time
+
+import numpy as np
+import netCDF4 as nc
+from eccodes import *
+from multiprocessing import Pool
+
+from IPython import embed as shell
+
```
Comment (fabiolrdiniz) [inline src/conventional/decode_bufr_LDM_raob.py:16]:
```suggestion
```

**Review by huishao-r — APPROVED** [review]
looks good to me. Thanks!

**Pull request merged**